packages/react-components/react-toast/stories/Toast/ToastLifecycle.stories.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+import {
+  Toaster,
+  useToastController,
+  Toast,
+  ToastTitle,
+  ToastBody,
+  ToastFooter,
+  ToastStatus,
+} from '@fluentui/react-toast';
+import { useId, Link, Button, Text, makeStyles, shorthands, tokens } from '@fluentui/react-components';
+
+const useStyles = makeStyles({
+  root: {
+    display: 'flex',
+    ...shorthands.gap('20px'),
+  },
+
+  button: {
+    display: 'block',
+  },
+
+  logContainer: {
+    display: 'flex',
+    flexDirection: 'column',
+  },
+
+  logLabel: {
+    color: tokens.colorNeutralForegroundOnBrand,
+    backgroundColor: tokens.colorBrandBackground,
+    width: 'fit-content',
+    fontWeight: tokens.fontWeightBold,
+    ...shorthands.padding('2px', '12px'),
+  },
+
+  log: {
+    overflowY: 'auto',
+    boxShadow: tokens.shadow16,
+    position: 'relative',
+    minWidth: '200px',
+    height: '200px',
+    ...shorthands.border('2px', 'solid', tokens.colorBrandBackground),
+    ...shorthands.padding('12px', '12px'),
+  },
+});
+
+export const ToastLifecycle = () => {
+  const styles = useStyles();
+  const toasterId = useId('toaster');
+  const labelId = useId();
+  const { dispatchToast } = useToastController(toasterId);
+  const [statusLog, setStatusLog] = React.useState<[number, ToastStatus][]>([]);
+  const [dismissed, setDismissed] = React.useState(true);
+  const notify = () => {
+    dispatchToast(
+      <Toast>
+        <ToastTitle intent="success" action={<Link>Undo</Link>}>
+          Email sent
+        </ToastTitle>
+        <ToastBody subtitle="Subtitle">This is a toast body</ToastBody>
+        <ToastFooter>
+          <Link>Action</Link>
+          <Link>Action</Link>
+        </ToastFooter>
+      </Toast>,
+      {
+        timeout: 1000,
+        onStatusChange: (e, { status: toastStatus }) => {
+          setDismissed(toastStatus === 'unmounted');
+          setStatusLog(prev => [...prev, [Date.now(), toastStatus]]);
```

Comment:
```suggestion
          setStatusLog(prev => [[Date.now(), toastStatus], ...prev]);
```

Let's use different order, so new entries will appear on top